internal/service/cluster/db/repo/repository.go

```diff
@@ -126,3 +128,64 @@ func (r *ClusterRepository) GetClusterResourcesNotIn(ctx context.Context, keys [
 func (r *ClusterRepository) GetClusterResource(ctx context.Context, id uuid.UUID) (*models.ClusterResource, error) {
 	return utils.Find[models.ClusterResource](ctx, r.Db, id)
 }
+
+// UpsertAlarmDefinitions inserts or updates alarm definition records
+func (r *ClusterRepository) UpsertAlarmDefinitions(ctx context.Context, records []models.AlarmDefinition) ([]models.AlarmDefinition, error) {
```

Comment:
As discussed, this is ok for now, but eventually these alarm dictionary objects will be subject to us having to publish notification events when they are added/updated/deleted so we will need to use the `Persist*` functions that were added to handle writing to the DB and capturing the data change event for notification purposes.